ui/common/ContributionBar.tsx

```diff
@@ -0,0 +1,72 @@
+'use client'
+
+import Image from 'next/image'
+
+interface ContributionInfo {
+  username: string
+  avatar: string
+  color: string
+  percentage: number
+  side: number
+  zindex: string
+}
+
+interface ContributionInfoProps {
+  contributionInfo: ContributionInfo[]
+}
+
+export default function ContributionBar({
+  contributionInfo,
+}: ContributionInfoProps) {
+  let value = 0
+  return (
+    <div className="relative h-[30px] w-full rounded-[5px] bg-black/20">
+      <div className="absolute z-50 flex h-full w-1/2 border-r-2 border-dotted border-white/50"></div>
+      {contributionInfo.map((info, i) => {
+        value += info.percentage
+        return (
+          <>
+            <div
+              className={`absolute h-full w-full overflow-hidden rounded-[5px] `}
+            >
+              <div
+                className={`absolute inset-0 flex transform-gpu border-r-2 border-black transition-transform`}
+                style={{
+                  transform: `translate3d(${
+                    info.side === 0 ? 100 - info.percentage : -(100 - value)
+                  }%,0,0)`,
+                  backgroundColor: `#${info.color}`,
+                  zIndex: info.zindex,
+                }}
+              />
+            </div>
+            <div
+              className="absolute inset-0 z-50 flex transform-gpu items-center justify-center transition-transform"
+              style={{
+                transform: `translate3d(${
+                  info.side === 0
+                    ? (100 - info.percentage) / 2
+                    : -(
+                        (100 -
+                          (value - info.percentage + info.percentage / 2) * 2) /
+                        2
+                      )
```

Comment:
This seems odd, could you explain what it does exactly?

Reply:
This is for determining the position at the which the avatar icon needs to be at with respect to the position of progress bar.
The default expression is:
`(100 - value) / 2`
let's say we have the `value = 20` which means 20 % of the progress bar.
The expression will give the value:
`(100 - 20) / 2 = 40`
Now, the avatar will be positioned on the centre after leaving an offset of 40 towards the right side.
<img width="1256" alt="Screenshot 2023-07-02 at 5 47 25 PM" src="https://github.com/saving-satoshi/saving-satoshi/assets/54861487/a43bd084-bb65-4131-b9b1-c5c4b53d2bd2">

The above expression works for a single avatar. Now, for multiple progress bar the expression becomes:
`[ 100 - {(PreviousValue + CurrentValue / 2) * 2}] / 2`
Basically the `value` is replaced by `{(PreviousValue + CurrentValue / 2) * 2}` 
Here is how the math for this one works:
<img width="465" alt="Screenshot 2023-07-02 at 5 52 28 PM" src="https://github.com/saving-satoshi/saving-satoshi/assets/54861487/b3b92c2b-ceff-4bcb-a274-b0fdfd3b5ef2">

Reply:
Thank's for the thorough explanation, while this works I think the following approach is a little cleaner and easier to understand: https://codesandbox.io/s/contribution-bar-5jy252?file=/src/ContributionBar.js I also took the liberty to rename some variables. It could use some further clean up (currently there's 2 loops doing basically the same thing) but I'll leave that up to you.

Reply:
Looks a bit overly complicated. You could just use a flex layout and set a `flex-basis` percentage on each box, with the last element having `margin-left: auto`, which pushes it to the right. Give the box `position:relative` and center the image with `position:absolute; left: 50%; top: 50%; transform: translate(-50%, -50%); z-index: 2;`. My 2 cents, you do what works best for you.